### ISSUE TITLE
All connections blacklisted, even after coming back online

### ISSUE THREAD

**Issue by jdowning** [taskrabbit/makara#36]
Similar to #32 

Environment:
- Rails 4.0.2
- Ruby 2.0.0
- PostgreSQL 9.2
- Makara 1dea24673e5a

Use case:
1. Start application with master and slave db online
2. GET generates 200
3. Stop slave db
4. GET generates 200
5. Start slave db _and_ stop master db
6. GET generates 500

The makara adapter doesn't appear to put the slave back into the pool when it comes back online and believes all connections are blacklisted:

```
ActionView::Template::Error ([Makara] All connections are blacklisted - [Makara] Blacklisted connection: master-db1 -> PG::ConnectionBad: PQconsumeInput() SSL connection has been closed unexpectedly

ActionView::Template::Error (undefined method `_makara_blacklisted?' for nil:NilClass):
```

**Comment by mnelson** [comment]
This may be fixed in master now that safe_value() does not assume a connection is present.

**Comment by jdowning** [comment]
This looks to be fixed in 72d3cc7. The only caveat is since the slaves are always trying to perform SELECTs. When the slave is taken down, there is always one failure (HTTP 500) before makara successfully switches to using the master db. (Refs #35)

**Comment by mnelson** [comment]
Closing. Merging of #35 will happen. Just trying to figure out these last few edge cases
